streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -999,7 +1006,9 @@ public synchronized Pattern sourceTopicPattern() {
         return this.topicPattern;
     }
 
-    public synchronized void updateSubscriptions(SubscriptionUpdates subscriptionUpdates) {
+    public synchronized void updateSubscriptions(SubscriptionUpdates subscriptionUpdates, String threadId) {
+        log.debug("stream-thread [{}] updating builder with {} topic(s) with possible matching regex subscription(s)", threadId, subscriptionUpdates);
```

Comment:
Tested with debug logging turned on and null threadId, prints `stream-thread [null]`